src/external/rawspeed/data/cameras.xml

```diff
@@ -3641,7 +3641,7 @@
 			<Hint name="coolpixmangled" value=""/>
 		</Hints>
 	</Camera>
-	<Camera make="NIKON" model="COOLPIX P7100" mode="12bit-compressed" decoder_version="1" supported="no"> <!-- need raw sample for whitelevel -->
+	<Camera make="NIKON" model="COOLPIX P7100" mode="12bit-uncompressed" decoder_version="1">
```

Comment:
Tested, works.
Ok, so either camera indeed writes uncompressed raw, or rawspeed's detection is borked in this case.